packages/shared/src/messages/MessageProxy.ts

```diff
@@ -0,0 +1,223 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+/* eslint-disable @typescript-eslint/no-explicit-any */
```

Comment:
should not be global for .ts files

Reply:
This is copied from aistudio directory, so I would rather it be fixed upstream on aistudio? 

This is used in order to proxy the messages back and forth from frontend to backend.

Reply:
regards to not using interface, it's because we abstract it and use a constructor / manipulate it: https://github.com/cdrage/podman-desktop-extension-bootc/blob/3903baba4c5ee7db88ba52c27520b73edf189e8d/packages/backend/src/api-impl.ts#L29 similar to aistudio's implementation.